conmon/conmon.c

```diff
@@ -212,9 +212,6 @@ ssize_t writev_buffer_append_segment(int fd, writev_buffer_t *buf, const void *d
 	if (data == NULL)
 		return 1;
 
-	if (len < 0)
```

Comment:
To protect ourselves from callers setting `len` < 0 (now that we no longer support it), I'd rather make `len` a `size_t` (vs. its current `ssize_t`) in `writev_buffer_append_segment`.  Then we can check for issues at compile-time with GCC's `-Wconversion` or similar.

Reply:
@wking Lets do that in another PR, and get this one merged.